src/main/java/com/aws/greengrass/smbridge/MessageBridge.java

```diff
@@ -97,16 +108,30 @@ private void handleMessage(Message message) {
                     LOGGER.atError().setCause(e).kv("Stream", stream).log("Stream Publish failed");
                 }
             };
-            // Perform topic matching on filter
+            // Perform topic matching on filter from mapped topics/destinations
             sourceDestinationMap.entrySet().stream().filter(
                     mapping -> MqttTopic.isMatched(mapping.getKey(), sourceTopic))
                     .map(Map.Entry::getValue)
                     .forEach(perTopicMapping -> perTopicMapping.forEach(processDestination));
+
+            // Perform topic matching against reserved topic
+            if (MqttTopic.isMatched(SMBridge.RESERVED_TOPIC, sourceTopic)) {
```

Comment:
What if there is a mapping as well for the reserve topic? We will end up sending the message twice?

Reply:
That is the current logic. From a design perspective, I think that's fine. Since the customer will be made aware of the reserved topic, they can configure it to go to many other destinations **if they want** (which might be a useful thing). On the flip side, it's not up to us to babysit their use of the reserved topic, because we've clearly documented it as reserved, so they use it at their own risk. What do you think?

Reply:
I can go either way I think. I am fine with a TODO in the code and defer the decision on this one. I don't think we need to resolve this in the context of this PR.